### PR TITLE
✨ Bento Carousel: "orientation" feature

### PR DESCRIPTION
**Pull request by caroqliu**
Partial for #28284, this PR introduces `orientation` as an attribute on `amp-base-carousel` and prop on `BaseCarousel`. [Demo available via PR deploy bot.](https://storage.googleapis.com/amp-test-website-1/amp_dist_83899/examples/storybook/preact/index.html?path=/story/basecarousel--default)

- `orientation`: Either `"horizontal"` or `"vertical"`, defaults to `"horizontal"`. When `"horizontal"` the carousel will lay out horizontally, with the user being able to swipe left and right. When `"vertical"`, the carousel lays out vertically, with the user being able to swipe up and down.
- This is a departure from `amp-base-carousel:0.1`, which allows the same configuration via [`horizontal="true"|"false"`](https://amp.dev/documentation/components/amp-base-carousel/#horizontal).

Open question:
- Should we show the carousel navigational arrows on top/bottom as opposed to left/right? It looks like vertically scrolling amp-base-carousel:0.1 has prev/next arrows on left and right. ([Demo](https://jsbin.com/yiguleyaku/edit?html,output)) As a user I might assume they would be top and bottom.

### CODE REVIEW

**Comment by dvoytenko** [comment]
> Should we show the carousel navigational arrows on top/bottom as opposed to left/right?

I'd think top/bottom would make the most sense. But can we do outset arrows that way in AMP without CLS?

File: extensions/amp-base-carousel/1.0/base-carousel.js [hunk 1]
```diff
@@ -95,9 +97,18 @@ function BaseCarouselWithRef(
   const setCurrentSlide =
     carouselContext.setCurrentSlide ?? setCurrentSlideState;
   const {slides, setSlides} = carouselContext;
+
   const scrollRef = useRef(null);
   const containRef = useRef(null);
   const contentRef = useRef(null);
+
+  const [axis, setAxis] = useState(
```
Comment (dvoytenko) [inline extensions/amp-base-carousel/1.0/base-carousel.js:105]: Is state/effect needed here? Seems like we can derive all necessary state from the `orientation` prop w/o an extra re-rendering step. No?

**Review by dvoytenko — CHANGES_REQUESTED** [review]

File: extensions/amp-base-carousel/1.0/base-carousel.js (hunk 1 above)
Comment (caroqliu) [reply extensions/amp-base-carousel/1.0/base-carousel.js:105]: Good catch, removed.

**Review by dvoytenko — APPROVED** [review]

**Comment by caroqliu** [comment]
> I'd think top/bottom would make the most sense. But can we do outset arrows that way in AMP without CLS?

I'll add this as an open task in the tracking issue to investigate and make the visual change if possible.

**Pull request merged**